### PR TITLE
feat: Multi filters support

### PR DESCRIPTION
**Pull request by sagarnaikjuspay**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [X] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [X] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/components/FilterUtils.res [hunk 1]
```diff
@@ -20,3 +20,72 @@ let updateURLWithDefaultFilter = (~path, ~filterParam, ~filterString) => {
     RescriptReactRouter.replace(finalUrl)
   }
 }
```
Comment (jainlokesh318) [inline src/components/FilterUtils.res:22]: Please discuss the approach once as its changing a lot of BaseComponents.

File: src/components/FilterUtils.res [hunk 2]
```diff
@@ -1,22 +1,68 @@
...
-    let strValue = RemoteFiltersUtils.getStrFromJson(key, val)
-    if strValue !== "" {
-      Some(`${key}=${strValue}`)
-    } else {
-      None
-    }
+let useFiltersObject = () => {
+  open HyperswitchAtom
+  filtersAtom
+  ->Recoil.useRecoilValueFromAtom
```
Comment (Riddhiagrawal001) [inline src/components/FilterUtils.res:4]: You can store a different type in Recoil other than string. Can be defined like 

Recoil.recoilAtom<Type> = Recoil.atom(.
  "recoil_name",
 initialValue,
)

Comment (sagarnaikjuspay) [reply src/components/FilterUtils.res:4]: we want string only

File: src/utils/UrlUtils.res [hunk 3]
```diff
@@ -36,34 +36,34 @@ let useGetFilterDictFromUrl = prefix => {
...
-      ->Js.String2.split("&")
-      ->Belt.Array.keepMap(str => {
-        let arr = str->Js.String2.split("=")
-        let key = arr->Belt.Array.get(0)->Belt.Option.getWithDefault("-")
-        let val = arr->Belt.Array.sliceToEnd(1)->Js.Array2.joinWith("=")
-        key === "" || val === "" ? None : Some((key, val))
-      })
-      ->Js.Dict.fromArray
-    let path = url.path->Belt.List.toArray->Js.Array2.joinWith("/")
+// let useUpdateUrlWith = (~prefix as _: string) => {
```
Comment (vsrivatsa-juspay) [inline src/utils/UrlUtils.res:39]: Remove commented code?

Comment (sagarnaikjuspay) [reply src/utils/UrlUtils.res]: removed it

File: src/screens/HyperSwitch/Analytics/SystemMetrics/SystemMetricsAnalytics.res [hunk 4]
```diff
@@ -74,6 +74,256 @@ module APITableInfo = {
...
+  @react.component
+  let make = () => {
+    let (_totalVolume, setTotalVolume) = React.useState(_ => 0)
+
+    let getStatData = (
+      singleStatData: systemMetricsObjectType,
+      timeSeriesData: array<systemMetricsSingleStateSeries>,
+      deltaTimestampData: DynamicSingleStat.deltaRange,
+      colType,
+      _mode,
```
Comment (vsrivatsa-juspay) [inline src/screens/HyperSwitch/Analytics/SystemMetrics/SystemMetricsAnalytics.res:91]: mode is unused in every case, can get rid of it

File: src/screens/HyperSwitch/Analytics/SystemMetrics/SystemMetricsAnalytics.res [hunk 5]
```diff
@@ -74,6 +74,256 @@ module APITableInfo = {
   }
 }
 
+module ConnectorLatency = {
```
Comment (vsrivatsa-juspay) [inline src/screens/HyperSwitch/Analytics/SystemMetrics/SystemMetricsAnalytics.res:77]: this module itself is unused and can be cleaned up if not required

**Review by vsrivatsa-juspay — DISMISSED** [review]

**Review by vsrivatsa-juspay — APPROVED** [review]

**Review by JeevaRamu0104 — COMMENTED** [review]
please recheck the functionality

File: src/context/FilterContext.res [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      setfilterDict(prev => {
+        let prevDictArr =
+          prev
+          ->Js.Dict.entries
+          ->Belt.Array.keepMap(
+            item => {
+              let (key, value) = item
+              switch dict->Js.Dict.get(key) {
+              | Some(_) => None
+              | None => value !== "" ? Some(item) : None
```
Comment (JeevaRamu0104) [inline src/context/FilterContext.res:51]: please use Js.String2.length

File: src/context/FilterContext.res [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+              }
+            },
+          )
+        let currentDictArr =
+          dict
+          ->Js.Dict.entries
+          ->Js.Array2.filter(
+            item => {
+              let (_, value) = item
+              value !== ""
```
Comment (JeevaRamu0104) [inline src/context/FilterContext.res:61]: same here

File: src/context/FilterContext.res [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        }
+        temp := dict->FilterUtils.parseFilterDict
+        dict
+      })
+    }
+
+    let reset = () => {
+      let dict = Js.Dict.empty()
+      setfilterDict(_ => dict)
+      temp := dict->FilterUtils.parseFilterDict
```
Comment (JeevaRamu0104) [inline src/context/FilterContext.res:79]: why temp is used?

**Review by JeevaRamu0104 — COMMENTED** [review]
image
Add trim in the payment_id search field

File: src/screens/HyperSwitch/Order/Orders.res [hunk 9]
```diff
@@ -81,6 +80,21 @@ let make = (~previewOnly=false) => {
...
+      searchVal=searchText
+      filterUrl
+      setFilters
+      endTimeFilterKey
+      startTimeFilterKey
+      initialFilters
+      initialFixedFilter
+      setOffset
+    />
+  }, [])
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/Order/Orders.res:96]: why empty array rather we can use React.useMemo0

File: src/screens/HyperSwitch/HSwitchRemoteFilter.res [hunk 10]
```diff
@@ -154,7 +131,7 @@ let useGetFiltersData = () => {
         }
       }
       None
-    }, [body])
+    }, [startTimeVal, endTimeVal, body->Js.Json.object_->Js.Json.stringify])
```
Comment (JeevaRamu0104) [inline src/screens/HyperSwitch/HSwitchRemoteFilter.res:134]: please re-check this useEffect1 shouldn't we use useEffect3

File: src/context/FilterContext.res (hunk 8 above)
Comment (sagarnaikjuspay) [reply src/context/FilterContext.res:79]: refactored it

**Review by vsrivatsa-juspay — APPROVED** [review]

**Review by vsrivatsa-juspay — DISMISSED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**